### PR TITLE
[CAS-20864] Patch the api-skeletons interface to allow package upgrades

### PR DESCRIPTION
**Pull request by el-toro-castoredc**
What the title says.

Without the patch, we'll get the following [error](https://app.circleci.com/pipelines/github/castoredc/edc/96438/workflows/2e0998f7-48aa-4e2f-96fb-ce05556fbb6c/jobs/1411134):
```
Fatal error: Declaration of ApiSkeletons\OAuth2\Doctrine\Adapter\DoctrineAdapter::setAuthorizationCode($code, $client_id, $user_id, $redirect_uri, $expires, $scope = null, $id_token = null) must be compatible with OAuth2\OpenID\Storage\AuthorizationCodeInterface::setAuthorizationCode($code, $client_id, $user_id, $redirect_uri, $expires, $scope = null, $id_token = null, $code_challenge = null, $code_challenge_method = null) in /app/vendor/api-skeletons/oauth2-doctrine/src/Adapter/DoctrineAdapter.php on line 703
```

This will also allow us to proceed with the update to PHP 8.2, although we'll have to update CDMS to use this repo again.

Also see https://github.com/castoredc/edc/pull/11903 and https://github.com/castoredc/edc/pull/11977/files

### CODE REVIEW

File: src/Adapter/DoctrineAdapter.php [hunk 1]
```diff
@@ -707,7 +707,9 @@ public function setAuthorizationCode(
         $redirect_uri,
         $expires,
         $scope = null,
-        $id_token = null
+        $id_token = null,
```
Comment (el-toro-castoredc) [inline src/Adapter/DoctrineAdapter.php:710]: Note that this is just an interface update to make things compatible. A proper fix would be to include the two new fields in the list of fields sent to the database, but that requires a lot more code tweaks. Without upstream changes (which most likely will never arrive), I think it's best to at least keep it compatible with the new interface from the `bshaffer/oauth2-server-php` package to not block other updates.

**Comment by oniric85** [comment]
Damn, sad we need to fork again. Do you think it could be worth contributing a PR to the original package?

**Comment by el-toro-castoredc** [comment]
> Damn, sad we need to fork again. Do you think it could be worth contributing a PR to the original package?

If we had a complete upgrade of the package, I'd happily submit it upstream. However, that's not the case, as it still has a lot more things to upgrade and consider, and that's a can of worms I'm not willing to open. In addition, the original repository has two [outstanding](https://github.com/API-Skeletons/oauth2-doctrine/pulls) PRs, one of which is from Apr 2022, so I'd count it as practically abandoned. I'd consider removing this package entirely and replacing it with something else, but given the current plans for CDMS, I'm not sure it's worth pursuing that path.

**Review by slknijnenburg — APPROVED** [review]
I think this is okay for now, LGTM to get 8.2 working. Longer term it might still be worthwhile to add the backend code for code_challenge and code_challenge_method as it adds support for PKCE, but that's not something we'll need tomorrow.

**Pull request merged**